File: src/main/java/de/thatsich/intellie/common/module/block/ABlock.java
package de.thatsich.intellie.common.module.block;

import de.thatsich.intellie.common.IInstantiate;
import de.thatsich.intellie.common.module.IName;
import de.thatsich.intellie.common.registries.BlockRegistry;
import net.minecraft.block.Block;
import net.minecraft.creativetab.CreativeTabs;


public abstract class ABlock extends Block implements IBlock, IInstantiate
{

	protected ABlock( ABlockInfo info, ABlockConfig config, BlockRegistry blocks )
	{
		super( info.getMaterial() );

		final IName blockName = info.getName();
		final String name = blockName.getUnlocalizedName();
		final float hardness = info.getHardness();
		final CreativeTabs creativeTab = info.getCreativeTab();
		final String texture = info.getTexture().getTexture();

		this.setBlockName( name );
		this.setHardness( hardness );
		this.setBlockTextureName( texture );
		this.setCreativeTab( creativeTab );

		blocks.add( this );
	}
}

File: src/main/scala/de/thatsich/minecraft/intellie/common/module/block/ABlock.scala
package de.thatsich.minecraft.intellie.common.module.block

import net.minecraft.block.Block
import net.minecraft.creativetab.CreativeTabs
import de.thatsich.minecraft.intellie.common.registries.BlockRegistry

/**
 *
 *
 * @author thatsIch
 * @since 07.04.2014.
 */
abstract class ABlock(info: ABlockInfo, config: ABlockConfig, blocks: BlockRegistry) extends Block(info.getMaterial) with IBlock
{
	val blockName: Nothing = info.getName
	val name: Nothing = blockName.getUnlocalizedName
	val hardness: Float = info.getHardness
	val creativeTab: CreativeTabs = info.getCreativeTab
	val texture: Nothing = info.getTexture.getTexture
	this.setBlockName(name)
	this.setHardness(hardness)
	this.setBlockTextureName(texture)
	this.setCreativeTab(creativeTab)
	blocks.add(this)
}
